internal/gitserver/addrs.go

```diff
@@ -0,0 +1,68 @@
+package gitserver
+
+import (
+	"crypto/md5"
+	"encoding/binary"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/gitserver/protocol"
+)
+
+var addrForRepoInvoked = promauto.NewCounterVec(prometheus.CounterOpts{
+	Name: "src_gitserver_addr_for_repo_invoked",
+	Help: "Number of times gitserver.AddrForRepo was invoked",
+}, []string{"user_agent"})
+
+// NewGitserverAddressesFromConf fetches the current set of gitserver addresses
+// and pinned repos for gitserver.
+func NewGitserverAddressesFromConf() GitServerAddresses {
+	cfg := conf.Get()
+	addrs := GitServerAddresses{
+		Addresses: cfg.ServiceConnectionConfig.GitServers,
+	}
+	if cfg.ExperimentalFeatures != nil {
+		addrs.PinnedServers = cfg.ExperimentalFeatures.GitServerPinnedRepos
+	}
+	return addrs
+}
+
+func newTestGitserverAddresses(addrs []string) GitServerAddresses {
+	return GitServerAddresses{
+		Addresses: addrs,
+	}
+}
+
+type GitServerAddresses struct {
```

Comment:
nit: to be consistent with other `gitserver` naming (without capital S)
```suggestion
type GitserverAddresses struct {
```

Reply:
Whoop, good call, thanks!